bfsbverify

```diff
@@ -0,0 +1,1061 @@
+o3SbK3TT1/4WOQ6YW/i36JaIFUyGZYYe2v0JYtxJoScxGv3Oj5kNK6J5Nun10Lkq
+8PcyxUgRbL6fokDjXK5u9D0CAwEAAQ==
+-----END PUBLIC KEY-----
+EOF
+
+        if [ "$(cmp $inkey $mfgrotpk)" ]; then
+            return 0
+        fi
+
+        # Otherwise try third key.
+        cat <<EOF > $mfgrotpk
+-----BEGIN PUBLIC KEY-----
+MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAuBJr0DLdcdrHcYWRE27c
+fHgfa6hOTfwfjLZOtuVMopTYCkhVWzxOc/FAh9s1oFK4j3BSpNr0G0SDUn4lJcPX
+AvJ0aDDAHvsAPPQSWDolDdz9AZ1glVVeeEzyaPl6aPm36nxjsEQSm4lA+63L7Y/C
+ndnEYr8aq8jiIrnxWbcNK+g+vNt5CL6+mLclyex0OqFJ644s349Klinwx9rhqaxu
+AGXgrj+C6ZTTBtZq1gswaFqeYt4nu+l6W90zEMwrcdhFHLFe4B5yZ+aimYl+6gKw
+iTjQ9C0uUeOgTWtu8gR65QJV0uddpoYshCXio8GaM0iXgq5R7sbVXVccijUT/h7r
+JJ3k3+W5/9iA3UrTgl/yrt89zCvbs8mYGItdf4vCzcVquNDyidLhpkOcn+th0PSs
++UVxwD5sJzHU4Agxrz/MZkVqrCdFXWaVD+BVlHH0kesD9JZ1x+LMfKujmpBrtNnC
+PXROuX+nLs/9llrMY22OfMmDC/f187jG/BIuo61TUZ8VcRAUQEQIl4I6e5UiWz6R
+IY4B6jyE7OQ5QU4ZvWT1H/C1Hor5Io6onmjqtIVFPt5R/qKHY7RWFGyyo8YfvaKd
+J6inGqrvKJZ4V2FzcRWS0PMUOe1YyOmhQrrgyP2Ts74geB90vxQ37p7AeuLlMbce
+uhHZD+izSMuQe3JqTkkLoxcCAwEAAQ==
+-----END PUBLIC KEY-----
+EOF
+
+        if [ "$(cmp $inkey $mfgrotpk)" ]; then
+            return 0
+        fi
+
+        # Otherwise try fourth key.
+        cat <<EOF > $mfgrotpk
+-----BEGIN PUBLIC KEY-----
+MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAtMMI8zlnuA2BSY4iGKX0
+vFgp41VGsKsUh3zhbsjcrpKUMpbCsIOjLMQ/m3zUTfJP9WjoFddI0rnPDJVkPZBf
+zzMtmrw+l28G8eM9/eSReEroPpg39g/wrw1Uk7xycI9nIL37F+Jfl4J7YxqoFAvG
+qmX3gxmXtFMtB61XKKHV7+RBwt7d1mMToPdpuJ1of8lztznh/hUrq5TFKugzhKhu
+BBptsJZE8hjLEt1JmZxxeF4RhhJgdoDa9RAO/5CebCmFEeqPvBDCTNzZmU64S3pS
+oW3JQVvi8y+w5hN54xnGdK/ZuG37E04cIUreuSAnDyqj107dRCAZi08rcnmdB3pF
+BhRUQscwqEOigxSbi44upn2AxWgY/I7AI/gq3Pp+JzCK0ClwmVkah5VxBZSVQ+BK
+rwGJC4Fw75FQWw/eprGRaEjlfKRsDwCqxeekk32APtjoQZt0l648S7IH0Jo/QsTM
+eqk9SWhuIYs96GbWzDlCo0HAxb8TLL1Ml5pifcjnjd7X9XaBTAERh0ejPQdZ06Az
+hcqfc/9++AZT6mZuiBu7ORMS33o6jGPt+oN9WDRNYYO96MWaTaPCROsWNR12okRO
+2n41NhVjCHuI332awo5TB5WBR6zuqQ+s3ZG4jwzmLGjfUmW6yz3sVgEkCI9/8thR
+oWOXTHe1tWmSg6xE9jkwD4ECAwEAAQ==
+-----END PUBLIC KEY-----
+EOF
+
+        if [ "$(cmp $inkey $mfgrotpk)" ]; then
+            return 0
+        fi
+
+    fi
+
+    touch $stamp
+    return 1
+}
+
+verify_rotpk()
+{
+    cert=$1
+    keyder=$2
+
+    certkey=${tmpdir}/pkey.pem
+    mfgkeystamp=${tmpdir}/mfgkeystamp
+    devkeystamp=${tmpdir}/devkeystamp
+
+    rm -f $certkey $mfgkeystamp $devkeystamp
+
+    # Extract certificate public key
+    openssl x509 -pubkey -noout -inform der -in $cert > $certkey
+
+    # Check if public key is a development key
+    compare_rotpk_dev $certkey $devkeystamp | true
+    # Check if public key is an official key
+    compare_rotpk_mfg $certkey $mfgkeystamp | true
+
+    if [ -f "$mfgkeystamp" ]; then
+        echo "  NVIDIA official ROT key (production)"
+    elif [ -f "$devkeystamp" ]; then
+        echo "  NVIDIA development ROT key (development)"
+    else
+        echo "  non-NVIDIA ROT key (unknown)"
+    fi
```

Comment:
Do we need to continue for unknown keys?

Reply:
I thought this script can be used by any BlueField user. When custom keys are used, I thought it is very useful for user to verify its own chain of trust. For that simple reason, I let the script continue. Another example would consist of using the script to test the certificate generation tool and/or secure boot validation on simulator/palladium, etc. Does it make sense?